src/cargo.rs

```diff
@@ -125,11 +130,16 @@ where
     /// ```
     ///
     /// [`Command`]: std::process::Command
+    #[deprecated(
+        since = "2.1.0",
+        note = "incompatible with a custom cargo build-dir, see instead `cargo::cargo_bin!`"
```

Comment:
I am confused by this deprecation notice. Does "custom cargo build-dir" mean custom `target` directory, e.g., as one would specify with `CARGO_TARGET_DIR`?

The incompatibility is not related to changing the `build` directory within the `target` directory (e.g., `target/debug/build`), correct?

If it is neither of these, could you please point me to a reference?

Reply:
Rust 1.91 split a `build-dir` out of `target-dir` with the default falling back to `target-dir`, for now though I hope to change that in the future.

See https://doc.rust-lang.org/cargo/reference/build-cache.html